lms/djangoapps/instructor/cybersource_enrollment_report.py

```diff
@@ -0,0 +1,172 @@
+        is_course_staff = has_access(user, 'staff', course)
+
+        # check the user enrollment role
+        if user.is_staff:
+            enrollment_role = _('Edx Staff')
+        elif is_course_staff:
+            enrollment_role = _('Course Staff')
+        else:
+            enrollment_role = _('Student')
+
+        course_enrollment = CourseEnrollment.get_enrollment(user=user, course_key=course_id)
+
+        if is_course_staff:
+            enrollment_source = _('Staff')
+        else:
+            # get the registration_code_redemption object if exists
+            registration_code_redemption = RegistrationCodeRedemption.registration_code_used_for_enrollment(
+                course_enrollment)
+            # get the paid_course registration item if exists
+            paid_course_reg_item = PaidCourseRegistration.get_course_item_for_user_enrollment(
+                user=user,
+                course_id=course_id,
+                course_enrollment=course_enrollment
+            )
+
+            # from where the user get here
+            if registration_code_redemption is not None:
+                enrollment_source = _('Used Registration Code')
+            elif paid_course_reg_item is not None:
+                enrollment_source = _('Credit Card - Individual')
+            else:
+                enrollment_source = _('Manually Enrolled')
+
+        enrollment_date = getattr(course_enrollment, 'created').strftime("%B %d, %Y")
+        currently_enrolled = getattr(course_enrollment, 'is_active')
+
+        course_enrollment_data = collections.OrderedDict()
+        course_enrollment_data['Enrollment Date'] = enrollment_date
+        course_enrollment_data['Currently Enrolled'] = currently_enrolled
+        course_enrollment_data['Enrollment Source'] = enrollment_source
+        course_enrollment_data['Enrollment Role'] = enrollment_role
+        return course_enrollment_data
+
+    def get_payment_info(self, user, course_id):
+        """
+        Returns the User Payment information.
+        """
+        course_enrollment = CourseEnrollment.get_enrollment(user=user, course_key=course_id)
+        paid_course_reg_item = PaidCourseRegistration.get_course_item_for_user_enrollment(
+            user=user,
+            course_id=course_id,
+            course_enrollment=course_enrollment
+        )
+        payment_data = collections.OrderedDict()
+        # check if the user made a single self purchase scenario
+        # for enrollment in the course.
+        if paid_course_reg_item is not None:
+            coupon_redemption = CouponRedemption.objects.select_related('coupon').filter(
+                order_id=paid_course_reg_item.order_id)
+            coupon_codes = [redemption.coupon.code for redemption in coupon_redemption]
+            coupon_codes = ", ".join(coupon_codes)
+
+            if coupon_redemption.exists():
+                list_price = paid_course_reg_item.list_price
```

Comment:
list price is stored in the `unit_cost` column, unless a coupon code was used, in which case the list price is stored in the `list_price` column and the discounted price is stored in the `unit_cost` column. if a coupon was not applied, the `list_price` is NULL.